qiskit_experiments/library/characterization/analysis/stark_ramsey_fast_analysis.py

```diff
@@ -0,0 +1,303 @@
+            },
+        )
+        ramsey_plotter.set_options(style=vis.PlotStyle({"figsize": (12, 5)}))
+
+        options = super()._default_options()
+        options.update_options(
+            result_parameters=[
+                curve.ParameterRepr("c1_pos", "stark_pos_coef_o1", "Hz"),
+                curve.ParameterRepr("c2_pos", "stark_pos_coef_o2", "Hz"),
+                curve.ParameterRepr("c3_pos", "stark_pos_coef_o3", "Hz"),
+                curve.ParameterRepr("c1_neg", "stark_neg_coef_o1", "Hz"),
+                curve.ParameterRepr("c2_neg", "stark_neg_coef_o2", "Hz"),
+                curve.ParameterRepr("c3_neg", "stark_neg_coef_o3", "Hz"),
+                curve.ParameterRepr("f_err", "stark_ferr", "Hz"),
+            ],
+            data_subfit_map={
+                "Xpos": {"series": "X", "direction": "pos"},
+                "Ypos": {"series": "Y", "direction": "pos"},
+                "Xneg": {"series": "X", "direction": "neg"},
+                "Yneg": {"series": "Y", "direction": "neg"},
+            },
+            plotter=ramsey_plotter,
+        )
+
+        return options
+
+    def _generate_fit_guesses(
+        self,
+        user_opt: curve.FitOptions,
+        curve_data: curve.CurveData,
+    ) -> Union[curve.FitOptions, List[curve.FitOptions]]:
+        """Create algorithmic initial fit guess from analysis options and curve data.
+
+        Args:
+            user_opt: Fit options filled with user provided guess and bounds.
+            curve_data: Formatted data collection to fit.
+
+        Returns:
+            List of fit options that are passed to the fitter function.
+        """
+        # Compute offset guess
+        user_opt.p0.set_if_empty(
+            offset=np.mean(curve_data.y),
+            f_err=0.0,
+        )
+        user_opt.bounds.set_if_empty(
+            offset=(-1, 1),
+            amp=(0, 1),
+        )
+        est_offs = user_opt.p0["offset"]
+
+        # Compute amplitude guess
+        amps = np.zeros(0)
+        for direction in ("pos", "neg"):
+            ram_x_off = curve_data.get_subset_of(f"X{direction}").y - est_offs
+            ram_y_off = curve_data.get_subset_of(f"Y{direction}").y - est_offs
+            amps = np.concatenate([amps, np.sqrt(ram_x_off**2 + ram_y_off**2)])
+        user_opt.p0.set_if_empty(amp=np.median(amps))
+        est_a = user_opt.p0["amp"]
+        d_const = user_opt.p0["dt"]
+
+        # Compute polynominal coefficients
+        guesses = []
+        for direction in ("pos", "neg"):
+            ram_x_data = curve_data.get_subset_of(f"X{direction}")
+            ram_y_data = curve_data.get_subset_of(f"Y{direction}")
+            xvals = ram_x_data.x
+
+            # Get normalized sinusoidals
+            xnorm = (ram_x_data.y - est_offs) / est_a
+            ynorm = (ram_y_data.y - est_offs) / est_a
+
+            # Compute derivative to extract polynominals from sinusoidal
+            dx = np.diff(xnorm) / np.diff(xvals)
```

Comment:
Minor: Should we guard more against the formatting of the data? We don't in other experiments like RamseyXY, so this would be something new. Here for example we assume that the X and Y data have the same amp points in the same order and that those amp points are ordered for the derivative. I wonder if we should sort by amp and possibly check that the amp points match for X and Y?

Reply:
Thanks. The amp equality check is added in [8dee4eb](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1009/commits/8dee4ebf3a645937466770b0ea50857af81aa2be). The value sorting is always done at base class (some other analyses such as RB also use derivative for initial guess).
https://github.com/Qiskit-Extensions/qiskit-experiments/blob/3511e4eae71042ef82bf59e0df7d315183836584/qiskit_experiments/curve_analysis/curve_analysis.py#L233-L239